tests/test-exports.c

```diff
@@ -361,15 +439,75 @@ test_full_context (void)
 
   exports = flatpak_context_get_exports (context, "com.example.App");
   g_assert_nonnull (exports);
+  print_exports (exports);
+
+  /* Most subdirectories of the root are exposed rw by "host" being in
+   * filesystems */
+  if (g_file_test ("/srv", G_FILE_TEST_EXISTS))
+    {
+      ExportedPath *ep = g_hash_table_lookup (exports->hash, "/srv");
+
+      g_assert_nonnull (ep);
+
+      if (g_file_test ("/srv", G_FILE_TEST_IS_SYMLINK))
+        g_assert_cmpint (ep->mode, ==, FAKE_MODE_SYMLINK);
+      else
+        g_assert_cmpint (ep->mode, ==, FLATPAK_FILESYSTEM_MODE_READ_WRITE);
+    }
+
+  /* This is usually a symlink on Debian systems, giving us an
+   * opportunity to test symlink handling */
+  if (g_file_test ("/vmlinuz", G_FILE_TEST_EXISTS))
+    {
+      ExportedPath *ep = g_hash_table_lookup (exports->hash, "/vmlinuz");
+
+      g_assert_nonnull (ep);
+
+      if (g_file_test ("/vmlinuz", G_FILE_TEST_IS_SYMLINK))
+        g_assert_cmpint (ep->mode, ==, FAKE_MODE_SYMLINK);
+      else
+        g_assert_cmpint (ep->mode, ==, FLATPAK_FILESYSTEM_MODE_READ_WRITE);
+    }
+
+  if (g_file_test ("/home", G_FILE_TEST_EXISTS))
+    {
+      ExportedPath *ep = g_hash_table_lookup (exports->hash, "/home");
+
+      g_assert_nonnull (ep);
+
+      if (g_file_test ("/home", G_FILE_TEST_IS_SYMLINK))
+        g_assert_cmpint (ep->mode, ==, FAKE_MODE_SYMLINK);
+      else
+        g_assert_cmpint (ep->mode, ==, FLATPAK_FILESYSTEM_MODE_READ_WRITE);
+    }
+
+  /* --filesystem=host takes precedence over --no-filesystem=/opt.
+   * TODO: Should it? */
```

Comment:
I added more test coverage while I was here, and found this, which doesn't look quite right.

Reply:
I dropped this part from the branch, because it doesn't really fit. I'm happy to revisit it in a separate PR.